### PR TITLE
1995 split cases collection

### PR DESCRIPTION
**Pull request by iamleeg**
This does all the stuff to make sure that GISAID/other excluded sources don't get into the cases collection, and reverts to using `find()` queries instead of aggregations for looking up cases which is sometimes hundreds of times faster.

What is _doesn't_ do is migrate any of the existing data: AFAICT the only impact is that there's GISAID data in dev so we can delete and re-import if needed. No impact on prod.

### CODE REVIEW

**Comment by iamleeg** [comment]
@abhidg @jim-sheldon fixed the date filtering now, this should be ready to go.

**Review by abhidg — COMMENTED** [review]
Not sure about the package-lock.json diff (lockfileVersion changed, maybe due to different npm version?) other than that, looks good.

File: data-serving/data-service/src/controllers/case.ts [hunk 1]
```diff
@@ -255,85 +209,46 @@ export class CasesController {
...
-                .allowDiskUse(true)
-                .collation({
-                    locale: 'en_US',
-                    strength: 2,
-                });
-            logger.info('Got results');
-            const docs = results[0].docs;
-            const total = results[0].total ?? 0;
+            const [docs, total] = await Promise.all([
+                sortedQuery.skip(limit * (page - 1)).limit(limit + 1),
```
[the remark this answers is not in the document]
Comment (iamleeg) [reply data-serving/data-service/src/controllers/case.ts:230]: I reverted this to the code that was there before the aggregation so this is what the backend already did.

[the remark this answers is not in the document]
Comment (iamleeg) [reply data-serving/data-service/src/controllers/case.ts:230]: @abhidg that behaviour was added in a massive commit: 389118bad. Unfortunately no comment or log message explaining the +1.

**Review by abhidg — APPROVED** [review]

**Review by jim-sheldon — APPROVED** [review]

**Pull request merged**